BrickKit/app/src/main/java/com/wayfair/brickkitdemo/StaggeredInfiniteScrollBrickFragment.java

```diff
@@ -86,4 +87,36 @@ private void addNewBricks(BrickPaddingFactory brickPaddingFactory) {
             dataManager.addLast(unusedBrick2);
         }
     }
+
+    private static class NpaStaggeredGridLayoutManager extends StaggeredGridLayoutManager {
+        /**
+         * Constructor for the NpaStaggeredGridLayoutManager.
+         *
+         * @param spanCount   the number of columns to use in the StaggeredGridLayoutManager.
+         * @param orientation the orientation of the StaggeredGridLayoutManager.
+         */
+        NpaStaggeredGridLayoutManager(int spanCount, int orientation) {
+            super(spanCount, orientation);
+        }
+
+        /**
+         * There is a bug in recyclerviews (March 2017) which causes them to load animations for views which don't yet exist.
+         * For us this is a race condition, it currently only occurs on the infinite brick pages.
+         *
+         * @return false
+         */
+        @Override
+        public boolean supportsPredictiveItemAnimations() {
+            return false;
+        }
+
+        @Override
+        public void onLayoutChildren(RecyclerView.Recycler recycler, RecyclerView.State state) {
+            try {
+                super.onLayoutChildren(recycler, state);
+            } catch (IndexOutOfBoundsException e) {
+                e.printStackTrace();
```

Comment:
should this be in there?  i.e. at least log to a logger....or a logging API or callback?

Reply:
Removed the whole class